### PR TITLE
feature: rust in yocto blog post - take1

### PR DESCRIPTION
**Pull request by sarfata**

### CODE REVIEW

File: _posts/2023-10-17-rust-in-yocto.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+---
+title: "Rust on Yocto: A Seamless Integration"
+description: "A look at the various ways to integrate Rust into Yocto"
```
Comment (martijnthe) [inline _posts/2023-10-17-rust-in-yocto.md:3]: maybe
```suggestion
description: "A look at the various ways to use Rust into Yocto"
```

File: _posts/2023-10-17-rust-in-yocto.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+## meta-rust-bin
+
+[`meta-rust-bin`](https://github.com/rust-embedded/meta-rust-bin/tree/master), a project of the rust-embedded group, emerges as a sought-after alternative for packaging Rust programs within Yocto. As an OpenEmbedded/Yocto layer, it uses official pre-built rust toolchains.
+
+Among its other merits are its frequent updates aligning with the latest Rust versions and its support for older Yocto versions, ranging [from dunfell to langdale](https://github.com/rust-embedded/meta-rust-bin/blob/ef9a765b0785ed9e964691ac1a5a302dd0a0ea62/conf/layer.conf#L13-L20).
+
+Previously, `meta-rust-bin` presented a `cargo` class, rendering it incompatible with `meta-rust`. The recommendation was to use a `BBMASK` statement to "hide" the built-in `cargo` and `rust` classes. Fortunately, this stumbling block was recently addressed with [the classname undergoing a change to `cargo_bin`](https://github.com/rust-embedded/meta-rust-bin/pull/136), enabling seamless usage of both the official Rust support and meta-rust-bin within the same Yocto project.
```
Comment (martijnthe) [inline _posts/2023-10-17-rust-in-yocto.md:93]: Nice, I hadn't seen that this got merged.

File: _posts/2023-10-17-rust-in-yocto.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+## Another Important Consideration: Binaries Size
+
+When delving into Rust for embedded systems, certain facets remain untouched in our discussion but merit attention in future deliberations.
+
+- **The Rust Standard Library**: A conspicuous drawback in the methods discussed is the static linking of the Rust standard library into each binary file. This addition can augment the file by several hundred kilobytes. An ingenious workaround might be the construction of `libstd-rs` as a dynamic library, allowing all the rust binaries within the system to share it. As Rust binaries become an integral part of your system image, this factor will gain prominence. None of the solutions presented today support this but they all mention it as a future improvement. Presently, our tactic involves dynamically linking (filesystem links) our three binaries (`memfaultd`, `memfaultctl`, and `memfault-core-handler`) to a singular binary. This unique binary alters its function based on the name used in the invocation, a strategy we fondly refer to as the `busybox` approach.
```
Comment (martijnthe) [inline _posts/2023-10-17-rust-in-yocto.md:145]:
```suggestion
- **The Rust Standard Library**: A conspicuous drawback in the methods discussed is the static linking of the Rust standard library into each binary file. This addition can augment the file by several hundred kilobytes. An ingenious workaround might be the construction of `libstd-rs` as a dynamic library, allowing all the rust binaries within the system to share it. As Rust binaries become an integral part of your system image, this factor will gain prominence. None of the solutions presented today support this but they all mention it as a future improvement. Presently, our tactic involves symlinking (filesystem links) our three programs (`memfaultd`, `memfaultctl`, and `memfault-core-handler`) to a singular binary. This unique binary alters its function based on the name used in the invocation, a strategy we fondly refer to as the `busybox` approach.
```

I wouldn't use "dynamically linking" there.

**Review by martijnthe — APPROVED** [review]

File: _posts/2023-10-17-rust-in-yocto.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
+---
+title: "Rust on Yocto: A Seamless Integration"
+description: "A look at the various ways to integrate Rust into Yocto"
+author: tsarlandie
+tags: [rust, yocto, linux, embedded, cross-compilation]
+---
+
+<!-- excerpt start -->
+
+At Memfault, our love affair with Rust began in late 2022. What drew us to Rust? Well, the typical allure of a modern programming language: an impressive type-system, memory safety without the constant jitters, efficient concurrency management, a thriving package ecosystem, and overwhelming support from our engineering team. To put it simply, we're smitten. Our journey with Rust has been nothing short of transformative, enabling rapid progress and leading us to conquer challenges we previously deemed ... non-trivial.
```
Comment (bahildebrand) [inline _posts/2023-10-17-rust-in-yocto.md:10]: This excerpt is on the long side. I would suggest limiting it to just this first paragraph.

**Review by bahildebrand — APPROVED** [review]
LGTM. If you could do the grammarly run through I would appreciate it

File: _posts/2023-10-17-rust-in-yocto.md (hunk 2 above)
Comment (geekigirlx) [reply _posts/2023-10-17-rust-in-yocto.md:93]: Let's crosslink to previous webinars to boost SEO rankings. Recommend placing this before the subscription and table of contents callouts. Thomas, up to you if it should be phrased in the third person (as it's currently structured), or if you'd like to change it to "Check out MY previous webinars..."

> 📺 **Watch the Webinar Recordings**
>
> Check out Thomas' previous webinars for more Embedded Linux discussions: [Over-the-Air Updates for Embedded Linux Devices](https://go.memfault.com/over-the-air-updates-embedded-linux-devices) and [Wrangling Penguins: Better Embedded Linux Monitoring and Debugging](https://go.memfault.com/better-embedded-linux-monitoring-debugging-memfault).

File: _posts/2023-10-17-rust-in-yocto.md (hunk 3 above)
Comment (geekigirlx) [reply _posts/2023-10-17-rust-in-yocto.md:145]: Same idea, as the webinar callout above but for recommended readings of previous posts. 

>📚**Recommended Reading**
>
>Fan of Rust? Check out our other Rust articles: [Asynchronous Rust on Cortex-M Microcontrollers](https://interrupt.memfault.com/blog/embedded-async-rust), [Rust for Low Power Digital Signal Processing](https://interrupt.memfault.com/blog/rust-for-digital-signal-processing), [From Zero to main(): Bare metal Rust](https://interrupt.memfault.com/blog/zero-to-main-rust-1).

File: _posts/2023-10-17-rust-in-yocto.md (hunk 2 above)
Comment (sarfata) [reply _posts/2023-10-17-rust-in-yocto.md:93]: added - I think the 3rd person makes it a little less "sales-y" so I kept it.

**Pull request merged**